generator/src/main/java/com/linkedin/pegasus/generator/DataSchemaParser.java

```diff
@@ -205,8 +208,8 @@ private void init(AbstractMultiFormatDataSchemaResolver resolver,
   public static class ParseResult
   {
     private static final String EXTENSION_FILENAME_SUFFIX = "Extensions.pdl";
-    // Sort the results to ensure that the output is deterministic for a given set of source inputs
-    private final Map<DataSchema, DataSchemaLocation> _schemaAndLocations = new TreeMap<>(Comparator.comparing(DataSchema::toString));
+    // Store the results in a LinkedHashMap to ensure ordering is deterministic for a given set of source inputs
```

Comment:
Is storing results in a hashmap giving the deterministic ordering? IMHO, the sortedSources gives the deterministic ordering.

Reply:
Correct, this map "maintains" the ordering from the original sort. (Since this map is later iterated upon)